code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -120,6 +120,71 @@
 	mute = MUTE_NONE
 	security_lock = TRUE
 	locked = FALSE
+	materials = list(MAT_METAL=500, MAT_GLASS=50)
+
+/obj/item/clothing/mask/muzzle/safety/shock
+	name = "shock muzzle"
+	desc = "A muzzle designed to prevent biting.  This one is fitted with a behavior correction system."
+	var/obj/item/assembly/trigger = null
+	origin_tech = "materials=1;engineering=1"
+	materials = list(MAT_METAL=500, MAT_GLASS=50)
+
+/obj/item/clothing/mask/muzzle/safety/shock/attackby(obj/item/W, mob/user, params)
+	if(isscrewdriver(W))
+		to_chat(user, "<span class='notice'>You disassemble [src].</span>")
+		trigger.forceMove(get_turf(user))
+		trigger.master = null
+		trigger.holder = null
+		trigger = null
+	else if(trigger)
+		to_chat(user, "<span class='notice'>Something is already attached to [src].</span>")
+	else if(istype(W, /obj/item/assembly/signaler) || istype(W, /obj/item/assembly/voice))

```

Comment:
You might want to add a message here when using other kinds of assemblies, that only voice or signaler assemblies are allowed.

Reply:
Added